utils/replicaset/replicaset.go

```diff
@@ -57,6 +64,69 @@ func FindNewReplicaSet(rollout *v1alpha1.Rollout, rsList []*appsv1.ReplicaSet) *
 	return nil
 }
 
+func CheckAndCreateDefaultAntiAffinityRule(RSTemplate *corev1.PodTemplateSpec, rollout v1alpha1.Rollout) {
+	enableAntiAffinity := (rollout.Spec.Strategy.BlueGreen != nil && rollout.Spec.Strategy.BlueGreen.AntiAffinity) || (rollout.Spec.Strategy.Canary != nil && rollout.Spec.Strategy.Canary.AntiAffinity)
+	if enableAntiAffinity && rollout.Status.StableRS != "" && rollout.Status.StableRS != rollout.Status.CurrentPodHash { //Check if flag set, not relevant for 1st RS
+		antiAffinityRule := GetDefaultAntiAffinityRule(rollout)
+		if RSTemplate.Spec.Affinity == nil {
+			RSTemplate.Spec.Affinity = &corev1.Affinity{}
+		}
+		if RSTemplate.Spec.Affinity.PodAntiAffinity == nil {
+			RSTemplate.Spec.Affinity.PodAntiAffinity = &corev1.PodAntiAffinity{}
+		}
+		if RSTemplate.Spec.Affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution == nil {
+			RSTemplate.Spec.Affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution = []corev1.PodAffinityTerm{antiAffinityRule}
+		} else {
+			exists := CheckIfDefaultAntiAffinityRuleExists(RSTemplate, rollout)
+			if exists == -1 {
+				RSTemplate.Spec.Affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution = append(RSTemplate.Spec.Affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution, antiAffinityRule)
+			}
+		}
+	}
+}
+
+func GetDefaultAntiAffinityRule(rollout v1alpha1.Rollout) corev1.PodAffinityTerm {
+	// Create anti-affinity rule for last stable rollout
+	antiAffinityRule := corev1.PodAffinityTerm{
+		LabelSelector: &metav1.LabelSelector{
+			MatchExpressions: []metav1.LabelSelectorRequirement{{
+				Key:      v1alpha1.DefaultRolloutUniqueLabelKey,
+				Operator: metav1.LabelSelectorOpIn,
+				// Most recent stable ReplicaSet
+				Values: []string{rollout.Status.StableRS},
+			}},
+		},
+		Namespaces:  []string{rollout.Namespace},
+		TopologyKey: "kubernetes.io/hostname",
+	}
+	return antiAffinityRule
+}
+
+func CheckIfDefaultAntiAffinityRuleExists(RSTemplate *corev1.PodTemplateSpec, rollout v1alpha1.Rollout) int {
+	defaultAntiAffinityRule := GetDefaultAntiAffinityRule(rollout)
+	if RSTemplate.Spec.Affinity != nil && RSTemplate.Spec.Affinity.PodAntiAffinity != nil && RSTemplate.Spec.Affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution != nil {
+		for i, podAffinityTerm := range RSTemplate.Spec.Affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution {
+			if cmp.Equal(podAffinityTerm, defaultAntiAffinityRule) {
```

Comment:
Can you use `reflect.DeepEqual` instead of using `cmp.Equal`? We have used `reflect.DeepEqual` in other parts of the codebase and then we will not need a new import.

Reply:
Note: this should revert the changes to the go path

Reply:
Actually, I don't think we need a DeepEqual comparsion because the antiAffinity changes with every stableRS change. We just need to check if the key is the `rollouts-pod-template-hash`